Error when using find_package(GTSAM) inside CMakeLists.txt

I cloned the latest develop branch of GTSAM on Ubuntu 16.04 and tried importing it in another [project](https://github.com/itzsid/test). It throws off an error when importing GTSAM. 

CMakeLists.txt looks like this: 
```
cmake_minimum_required(VERSION 3.0.0)
project(gtsam_test)
find_package(GTSAM REQUIRED)
```

Error is in `find_dependency()` inside GTSAMConfig.cmake. The error is as follows:
```
-- The C compiler identification is GNU 5.5.0
-- The CXX compiler identification is GNU 5.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /usr/share/cmake-3.5/Modules/CMakeFindDependencyMacro.cmake:45 (message):
  Invalid arguments to find_dependency
Call Stack (most recent call first):
  /usr/local/lib/cmake/GTSAM/GTSAMConfig.cmake:17 (find_dependency)
  CMakeLists.txt:3 (find_package)
```

Seems to be related to this [issue](https://github.com/RobustFieldAutonomyLab/LeGO-LOAM/issues/56) and this [issue](https://github.com/pabloariasal/modern-cmake-sample/issues/5).

--- a/cmake/Config.cmake.in
+++ b/cmake/Config.cmake.in
@@ -14,7 +14,12 @@ endif()
 
 # Find dependencies, required by cmake exported targets:
 include(CMakeFindDependencyMacro)
+# Allow using cmake < 3.8
+if(${CMAKE_VERSION} VERSION_LESS "3.8.0") 
+find_package(Boost @BOOST_FIND_MINIMUM_VERSION@ COMPONENTS @BOOST_FIND_MINIMUM_COMPONENTS@)
+else()
 find_dependency(Boost @BOOST_FIND_MINIMUM_VERSION@ COMPONENTS @BOOST_FIND_MINIMUM_COMPONENTS@)
+endif()
 
 # Load exports
 include(${OUR_CMAKE_DIR}/@PACKAGE_NAME@-exports.cmake)

